dataquality/integrations/fastai.py

```diff
@@ -0,0 +1,370 @@
+        :param old_func: The original function to patch.
+        :param store: The store to store the indices in.
+        """
+        self.old_func = old_func
+        self.store = store
+        self.store[FastAiKeys.dataloader_indices] = []
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        """
+        Call the original function and store the indices.
+        :param args: The arguments to pass to the original function.
+        :param kwargs: The keyword arguments to pass to the original function.
+        """
+        res = self.old_func(*args, **kwargs)
+        if res:
+            self.store[FastAiKeys.dataloader_indices].append(res)
+        return res
+
+
+class FastAiDQCallback(Callback):
+    """
+    Dataquality logs the model embeddings and logtis to measure the quality
+    of the dataset. Provide the label names and the classifier layer to log
+    the embeddings and logits. If no classifier layer is provided,
+    the last layer of the model will be used.
+    Here is how to take the last layer of the model:
+    `dqc = DataqualityCallback(labels=['negative','positive'], layer=model.fc)`
+    End to end example:
+    ```python
+    from fastai.vision.all import *
+    from fastai.callback.galileo import DataqualityCallback
+    path = untar_data(URLs.PETS)/'images'
+    image_files = get_image_files(path)#[:107]
+    label_func = lambda x: x[0].isupper()
+    dls = ImageDataLoaders.from_name_func(
+        path, image_files, valid_pct=0.2,
+        label_func=label_func, item_tfms=Resize(224),
+        num_workers=1)
+    learn = vision_learner(dls, 'resnet34', metrics=error_rate)
+    dqc = DataqualityCallback(labels=["nocat","cat"])
+    learn.add_cb(dqc)
+    learn.fine_tune(2)
+    ```
+    """
+
+    hook = None
+    is_initialized = False
+    labels = None
+    model_outputs_log: Dict[FastAiKeys, Any]
+    current_idx: List[int]
+    logger_config: BaseLoggerConfig
+    idx_store: Dict[FastAiKeys, List[Any]]
+
+    def __init__(
+        self,
+        labels: List[str],
+        layer: Any = None,
+        log_dataset: bool = True,
+        task_type: str = "image_classification",
+        options: Dict[str, Any] = {},
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Dataquality logs the model embeddings and logits to measure the quality
+        of the dataset. This helps to find mislabelled samples in a
+        data centric approach.
+        :param layer: Classifier layer with embeddings as input and logits as output.
+        :param log_dataset: Enable automatic extraction of the dataset to data quality.
+        :param disable_dq: Disable data quality logging.
+        :param args: The arguments to pass to the super class.
+        :param kwargs: The keyword arguments to pass to the super class.
+        """
+        super().__init__(*args, **kwargs)
+        self.disable_dq = os.environ.get("DQ_NOOP", False)
```

Comment:
we have a global helper for this already, please use that https://github.com/rungalileo/dataquality/blob/main/dataquality/utils/helpers.py#L30

```suggestion
        self.disable_dq = galileo_disabled()
```

Reply:
good catch